src/content/docs/identityserver/ui/login/dynamicproviders.md

```diff
@@ -124,21 +226,33 @@ public class CustomConfig : IConfigureNamedOptions<OpenIdConnectOptions>
 }
 ```
 
-And to register this in the DI system:
+You will need to register the named options type in the service container at startup:
 
-```cs
+```csharp title="Program.cs"
 builder.Services.ConfigureOptions<CustomConfig>();
 ```
 
-#### Accessing OidcProvider Data In IConfigureNamedOptions
+:::note
+In your `IConfigureNamedOptions<OpenIdConnectOptions>`, you can use constructor injection to access other services.
+For example, if you need to access your Entity Framework Core database context to retrieve additional data for a given
```

Comment:
This is actually a problem, because db access usually is async.

We should propose something different. Database access happens somewhere else, the result is cached. The cached data is read here. Something like that. 

Will need something similar for the BFF

Reply:
Reworded a bit, what do you think?